File: Assignment12/program4.c
/////////////////////////////////////////////////////////////////// 
//
// File name : program4.c
// Description : Pattern
// input : iRow = 3  iCol = 4
// Output : 
/*
    *   #   *   #
    *   #   *   #
    *   #   *   #
*/
// Author : Adinath Santosh Pawar
// Date : 27/5/25
// 
///////////////////////////////////////////////////////////////////

#include<stdio.h>

void Pattern(int iRow,int iCol)
{
    int j = 0;
    int i = 0;

    for(i = 1; i <= iRow; i++)
    {    
        for(j = 1; j <= iCol; j++)
        {
            if( (j % 2) == 0)
            {
                printf("# \t");
            }
            else
            {
                 printf("* \t");
            }
        }
        printf("\n");
    }

}

int main()
{
    int iValue1 = 0, iValue2 = 0;

    printf("enter number of rows \n");
    scanf("%d",&iValue1);

    printf("enter number of columns \n");
    scanf("%d",&iValue2);

    Pattern(iValue1,iValue2);
    return 0;
}

File: Assignment45/program4.java
import java.util.*;

/////////////////////////////////////////////////////////////////// 
//
// File name : program4.java
// Description : Check is matrix is identity maatrix
// Author : Adinath Santosh Pawar
// Date : 2/8/25
// 
///////////////////////////////////////////////////////////////////

class Matrix
{
    private int Arr[][];

    public Matrix(int A, int B)
    {
        Arr = new int[A][B];
    }
        
    public void Accept()
    {
        System.out.println("Enter the elements of matrix ");

        Scanner sobj = new Scanner(System.in);

        int i = 0, j = 0;
        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                Arr[i][j] = sobj.nextInt();
            }
        }
    }

    public boolean ChkIdentity(int iRow, int iCol) 
    {

        if (iRow != iCol) 
        {
            return false;
        }

        for (int i = 0; i < iRow; i++) 
        {
            for (int j = 0; j < iCol; j++) 
            {
                if (i == j && Arr[i][j] != 1) 
                {
                    return false;
                } 
                else if(i != j && Arr[i][j] != 0) 
                {
                    return false;
                }
            }
        }

        return true; 
    }


    public void Display()
    {
        System.out.println("The elements of matrix are ");

        int i = 0, j = 0;
        for(i = 0; i < Arr.length; i++)
        {
            for(j = 0; j < Arr[i].length; j++)
            {
                System.out.print(Arr[i][j] + " ");
            }
            System.out.println();
        }
    }

}
class program4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        boolean bRet = false;

        System.out.println("Enter number of rows :");
        int iRow = sobj.nextInt();

        System.out.println("Enter number of columns :");
        int iCol = sobj.nextInt();

        Matrix mobj = new Matrix(iRow,iCol);

        mobj.Accept();
        bRet = mobj.ChkIdentity(iRow,iCol);
        System.out.println(bRet);
    }
}